test/http.js

```diff
@@ -110,6 +120,40 @@ test('response contains url', async t => {
 	t.is((await got(s.url)).url, `${s.url}/`);
 });
 
+test('cancel in-progress request', async t => {
+	const p = got(`${s.url}`, {
+		// Readable stream we never close
+		body: new Readable({
+			read() {}
+		}),
+		timeout: 4000,
+		retries: 0
+	});
+
+	setTimeout(() => {
+		p.cancel();
+	}, 100);
+	await t.throws(p, PCancelable.CancelError);
+});
+
+test('cancel immediately', async t => {
+	const p = got(`${s.url}/store-id`);
+	p.cancel();
+	await t.throws(p, PCancelable.CancelError);
+
+	await delay(100);
+	t.false(cancelRequestCompleted);
+});
+
+test.failing('cancel throws after request completed', async t => {
```

Comment:
I did a little testing and a little thinking. Trying to resolve an already settled promise seems to do nothing in the case of Node too. A note in the documentation is enough.
I think you're right on this.